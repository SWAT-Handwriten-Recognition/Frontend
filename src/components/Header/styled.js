import styled from 'styled-components';

export const HeaderContainer = styled.header`
  align-items: center;
  background-color: black;
  display: flex;
  height: 20vh;
  justify-content: space-between;

  @media (max-width:425px){
    flex-direction: column;
    height: 30vh;
  }
`;
export const LogoContainer = styled.div``;
export const ButtonsContainer = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 1.5rem;
  width: 20%;

  @media (max-width: 1024px) {
    width: 30%;
  }
  @media (max-width: 425px) {
    width: 70%;
  }
`;
export const Button = styled.button`
  box-shadow: 3px 4px 0px 0px rgba(255, 255, 255, 5);
  background: linear-gradient(
    to bottom,
    rgba(255, 255, 255, 0) 5%,
    rgba(255, 255, 255, 1) 100%
  );
  background-color: whitesmoke;
  border-radius: 18px;
  border: 1px solid black;
  display: inline-block;
  cursor: pointer;
  color: black;
  font-family: Arial;
  font-size: 17px;
  padding: 7px 25px;
  text-decoration: none;
  transition: .3s;

  &:hover {
    background: linear-gradient(
      to bottom,
      rgba(255, 255, 255, 1) 5%,
      rgba(0, 0, 0, 1) 100%
    );
    box-shadow: 3px 4px 0px 0px rgba(255, 255, 255, 0.1);
    color: whitesmoke;
  }
  &:active {
    position: relative;
    top: 1px;
  }
`;
