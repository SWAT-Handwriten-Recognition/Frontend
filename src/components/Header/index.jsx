/**
 * This is header component show user login or sign up option
 */
import { HeaderContainer, LogoContainer, ButtonsContainer, Button } from './styled'
import { useHistory } from 'react-router-dom'
import Logo from '../../assets/Logo/index.jsx'

const Header = ({ scroll }) => {
  const history = useHistory()

  const handleSignUp = () => history.push('/signup')

  return (
    <HeaderContainer scroll={scroll}>
      <LogoContainer>
        <Logo />
      </LogoContainer>
      <ButtonsContainer>
        <Button>Sign In</Button>
        <Button onClick={handleSignUp}>Sign Up</Button>
      </ButtonsContainer>
    </HeaderContainer>
  )
}

export default Header
