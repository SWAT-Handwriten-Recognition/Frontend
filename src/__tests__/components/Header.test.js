import { shallow } from 'enzyme';
import Header from '../../components/Header/index.jsx';
import toJson from 'enzyme-to-json';

describe('<Header />', () => {
  const wrapper = shallow(<Header />);

  it('is the Header rendering?', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('matches the snap shot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
