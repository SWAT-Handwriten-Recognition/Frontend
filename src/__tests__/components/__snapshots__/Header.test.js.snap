// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches the snap shot 1`] = `
<styled__HeaderContainer>
  <styled__LogoContainer>
    <Logo />
  </styled__LogoContainer>
  <styled__ButtonsContainer>
    <styled__Button>
      Sign in
    </styled__Button>
    <styled__Button>
      Sign up
    </styled__Button>
  </styled__ButtonsContainer>
</styled__HeaderContainer>
`;
